com.unity.render-pipelines.high-definition/Runtime/Lighting/VolumetricLighting/VolumetricLighting.cs

```diff
@@ -258,18 +258,35 @@ static internal Vector3Int ComputeVolumetricViewportSize(HDCamera hdCamera, ref
 
             int   viewportWidth  = hdCamera.actualWidth;
             int   viewportHeight = hdCamera.actualHeight;
-            float screenFraction = controller.screenResolutionPercentage.value * 0.01f;
-            int   sliceCount     = controller.volumeSliceCount.value;
+
+            float screenFraction;
+            int   sliceCount;
+            if (controller.fogControlMode.value == FogControl.Balance)
+            {
+                // Evaluate the ssFraction and sliceCount based on the control parameters
+                float maxScreenSpaceFraction = (1.0f - controller.resolutionDepthRatio.value) * (Fog.maxFogScreenResolutionPercentage - Fog.minFogScreenResolutionPercentage) + Fog.minFogScreenResolutionPercentage;
+                screenFraction = Mathf.Lerp(Fog.minFogScreenResolutionPercentage, maxScreenSpaceFraction, controller.volumetricFogBudget.value) * 0.01f;
+                float maxSliceCount = Mathf.Max(1.0f, controller.resolutionDepthRatio.value * Fog.maxFogSliceCount);
+                sliceCount = (int)Mathf.Lerp(1.0f, maxSliceCount, controller.volumetricFogBudget.value);
```

Comment:
I find this math a little misleading.
The cost of volumetric lighting is O(n), where n = w*h*d is the total resolution of the buffer.
It appears that this math does not keep 'n' constant, since it interpolates both 'sliceCount' and 'screenFraction' linearly, while the latter has a quadratic impact on the overall resolution.

Reply:
Good point. However, while doing the implementation i found that the quadratic XY is more intuitive when raising the slider than a linear cost. Maybe we can merge as is and have a seperate branch  to make them try both and see which one feels better for and go for that one?

Reply:
yup, see in next PR